internal/consumer/consumer_test.go

```diff
@@ -0,0 +1,130 @@
+package consumer
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/JakeHumphries/gymshark-hacker-news/internal/models"
+	"github.com/stretchr/testify/mock"
+)
+
+type MockWriter struct {
+	mock.Mock
+}
+
+func (m *MockWriter) SaveItem(ctx context.Context, item models.Item) (*models.Item, error) {
+	m.Called()
```

Comment:
Function parameters should be passed to the mock's `Called` function. https://pkg.go.dev/github.com/stretchr/testify/mock#hdr-Example_Usage